### PR TITLE
Initial codeclimate config

### PR DESCRIPTION
**Pull request by edgibbs**

### CODE REVIEW

File: .eslintrc-codeclimate.js [hunk 1]
```diff
@@ -0,0 +1,35 @@
+// Maintained to handle codeclimate's non-support of some eslint rules
```
Comment (eckdanny-osi) [inline .eslintrc-codeclimate.js]: CC only _supports_ certain plugins!? [lame but true](https://docs.codeclimate.com/v1.0/docs/eslint#section-supported-plugins) 💩

File: .eslintrc-codeclimate.js [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+module.exports = {
+  extends: [
```
Comment (eckdanny-osi) [inline .eslintrc-codeclimate.js]: May be better to _extend_ the normal config and disable some stuff instead of forking configs.

**Comment by eckdanny-osi** [comment]
Is this a `WiP` or ready to be merged?

**Comment by edgibbs** [comment]
It's a WIP there are a few failing Code Climate things that need cleanup.  So any concrete ideas on extending the config.  I went down that path and then realized it wasn't as easy as I expected.

**Comment by edgibbs** [comment]
Build triggered. sha1 is original commit.

**Comment by edgibbs** [comment]
Build started sha1 is original commit.

**Comment by edgibbs** [comment]
Everything is green!

**Comment by edgibbs** [comment]
Build triggered. sha1 is merged.

**Comment by edgibbs** [comment]
Build started sha1 is merged.

**Comment by edgibbs** [comment]
Everything is green!

**Comment by edgibbs** [comment]
OK, there are lots of small changes now that were needed for codeclimate including adding Rubocop.

File: .eslintrc-codeclimate.js [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+    },
+  },
+  env: {
+    es6: true,
+    node: true,
+    'jest/globals': true,
+  },
+  rules: {
+    'prettier/prettier': 'error',
+    // 'no-unsanitized/method': 'error',
```
Comment (mlipardo) [inline .eslintrc-codeclimate.js]: Please remove comments

Comment (edgibbs) [reply .eslintrc-codeclimate.js]: These were removed by a later commit, but good catch.

**Pull request merged**